src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,115 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Booking;
+import com.bootcamp.demo.model.PaymentStatus;
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.model.component.Location;
+import com.bootcamp.demo.model.component.ScooterStatus;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.logging.Logger;
+
+@Service
+public class BookingService implements IBookingService {
+    private static final Logger LOGGER = Logger.getLogger(BookingService.class.getName());
+    private static final String COLLECTION_PATH = "bookings/databases/bookings";
+
+    private final Firestore db;
+
+    public BookingService(Firestore db) {
+        this.db = db;
+    }
+
+    /**
+     * Retrieves a Booking entity by its id.
+     */
+    @Override
+    public Booking getBookingByID(final UUID bookingId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (bookingId == null) throw new IllegalArgumentException();
+        DocumentReference documentReference = db.collection(COLLECTION_PATH)
+                .document(bookingId.toString());
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot document = future.get();
+        Booking booking;
+        if (document.exists()) {
+            booking = document.toObject(Booking.class);
+            return booking;
+        }
+        return null;
+    }
+
+    /**
+     * Retrieves a Booking entity by its user's id.
+     */
+    @Override
+    public LinkedHashSet<Booking> getBookings(final UUID userId) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (userId == null) throw new IllegalArgumentException();
+        LinkedHashSet<Booking> userBookings;
+        userBookings = db.collection(COLLECTION_PATH)
+                .whereEqualTo("accountId", userId).get().get()
+                .getDocuments()
+                .stream().map(d -> d.toObject(Booking.class))
+                .collect(Collectors.toSet())
+                .stream().sorted(Comparator.comparing(Booking :: getStartDate))
```

Comment:
There is no need to call stream.collect.stream (with the intermediate collect), as it will be the same as the following:
```
.stream()
.map(d -> d.toObject(Booking.class))
.sorted(Comparator.comparing(Booking :: getStartDate))
```
Same applies for `getAllBookings`.